internal/host/host.go

```diff
@@ -1050,11 +1050,14 @@ func (m *Manager) reportValidationStatusChanged(ctx context.Context, vc *validat
 					m.metricApi.HostValidationChanged(models.HostValidationID(v.ID))
 					eventgen.SendHostValidationFailedEvent(ctx, m.eventsHandler, *h.ID, h.InfraEnvID, h.ClusterID,
 						hostutil.GetHostnameForMsg(h), v.ID.String())
-				}
-				if v.Status == ValidationSuccess && currentStatus == ValidationFailure {
+				} else if v.Status == ValidationSuccess && currentStatus == ValidationFailure {
 					log.Infof("Host %s: validation '%s' is now fixed", hostutil.GetHostnameForMsg(h), v.ID)
 					eventgen.SendHostValidationFixedEvent(ctx, m.eventsHandler, *h.ID, h.InfraEnvID, h.ClusterID,
 						hostutil.GetHostnameForMsg(h), v.ID.String())
+				} else if v.Status != currentStatus {
```

Comment:
is this code just for host validations, or does it also work for cluster validations?

Reply:
Just host validatoins, need to think/check if we have the same issue there.

Reply:
Couldn't hurt to have it for cluster as well... It's hard to come up with / notice cases where these silent events are needed, to save future headache I think we should take care of cluster as well

Reply:
OK, from the code it seems that we might have the same problem there.
I'll add it